test/LibraryManager.IntegrationTest/FileSaveRestoreTests.cs

```diff
@@ -0,0 +1,344 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Web.LibraryManager.IntegrationTest
+{
+    [TestClass]
+    public class FileSaveRestoreTests : VisualStudioLibmanHostTest
+    {
+        ProjectTestExtension _webProject;
+        ProjectItemTestExtension _libManConfig;
+        const string _projectName = @"TestProjectCore20";
+
+        [TestInitialize()]
+        public void initialize()
+        {
+            _webProject = Solution.ProjectsRecursive[_projectName];
+            _libManConfig = _webProject.Find(SolutionItemFind.FileName, "libman.json");
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            int count = Editor.Edit.UndoStack.Count;
+            Editor.Edit.Undo(count);
+
+            _libManConfig.Save();
+
+            // Delete restored libraries
+            string deletePath = Path.Combine(SolutionRootPath, _projectName, "wwwroot", "lib");
+            Directory.Delete(deletePath, true);
+        }
+
+        [TestMethod]
+        public void FileSaveRestore_Cdnjs_AddNewLibraryWithoutSpecifingFiles()
+        {
+            _libManConfig.Open();
+            string[] expectedFilesAndFolders = new[] {
+                "jquery.js",
+                "jquery.min.js",
+            };
+
+            Editor.Caret.MoveToExpression("\"version\": \"1.0\"");
+            Editor.Caret.MoveToEndOfLine();
+            Editor.KeyboardCommands.Enter();
+            Editor.KeyboardCommands.Type("\"defaultProvider\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "cdnjs", caseInsensitive: true);
+            Editor.KeyboardCommands.Type("cdnjs");
+
+            Editor.Caret.MoveToExpression("\"libraries\"");
+            Editor.Caret.MoveDown(2);
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true, timeout: 5000);
+            Editor.KeyboardCommands.Type("jquery@3.3.1");
+            Editor.KeyboardCommands.Right();
```

Comment:
Why do you need this line?  Why can't we type something from jquery through the comma?

Reply:
Looks like there's a bug. If have comma in the line before, there will be two commas!

---
In reply to: [194552816](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194552816) [](ancestors = 194552816)

Reply:
get rid of it in new iteration.

---
In reply to: [194895044](https://github.com/aspnet/LibraryManager/pull/113#discussion_r194895044) [](ancestors = 194895044,194552816)